docs/user_guide/building_blocks/categories.md

```diff
@@ -1,3 +1,495 @@
+function that will do most of the work for us:
+
+```python
+def name_fn(root_cats):
+    """ define how to combine names for category combinations """
+    cat_name = "__".join(cat.name for cat in root_cats.values())
+    return cat_name
+
+
+def kwargs_fn(root_cats):
+    """ define how to combine category attributes for category combinations """
+    kwargs = {
+        "id": sum([c.id for c in root_cats.values()]),
+        "label": ", ".join([c.name for c in root_cats.values()]),
+        # optional: store the information on how the combined category has been created
+        "aux": {
+            "root_cats": {key: value.name for key, value in root_cats.items()},
+        },
+    }
+    return kwargs
+
+#
+# define how to combine categories
+#
+
+
+category_blocks = OrderedDict({
+    "lep": [config.get_category(f"{n_jet}lep") for n_lep in lep_categories],
+    "jet": [config.get_category(f"{n_jet}jet") for n_jet in jet_categories],
+    "ht": [config.get_category(f"ht{ht_down}to{ht_up}") for ht_down, ht_up in ht_categories],
+})
+
+from columnflow.config_util import create_category_combinations
+
+# this function creates all possible category combinations without producing duplicates
+# (e.g. category '1lep__2jet__ht200to400' will be produced, but '2jet__1lep__ht200to400' will be skipped)
+n_cats = create_category_combinations(
+    config,
+    category_blocks,
+    name_fn=name_fn,
+    kwargs_fn=kwargs_fn,
+    skip_existing=False,
+)
+
+# let's check what categories have been produced
+print(f"{n_cats} have been created by the create_category_combinations function")
+all_cats = [cat.name for cat, _, _ in config.walk_categories()]
+print(f"List of all cateogries in our config: \n{all_cats}")
+```
+
+To test our final set of categories, we can call our `CreateYieldsTable` task again.
+
+```shell
+law run cf.CreateYieldsTable --version v1 \
+    --calibrators example --selector example --producer category_ids \
+    --processes tt,st --categories "*"
+```
+
+
+:::{note} This categorization is not inclusive!
+
+Since we only store leaf category ids, all the events that are not considered by one of the category
+blocks will not be considered at all. For this set of categories it means that we implicitly added cuts
+on the number of jets (>= 2 and <= 6), the number of leptons (>= 1 and <= 3), and HT (<= 600) as soon
+as we use one of these categories.
+
+TODO: it might be more instructive to build this example such that our categorization is inclusive.
+:::
+
+
+## Extend your categorization as part of a Producer
+
+Some of your categories might need some computationally expensive reconstructed observables
+and can therefore only be called after certain requirements are met, or be called as part of a Producer.
+This can be achieved by adding category instances as part of the `Producer.init`. Columns from
+custom requirements can be added to a Producer via the `Producer.requires` in combination
+with the `Producer.setup`.
```

Comment:
```suggestion
and can therefore only be called after certain requirements are met.
In these cases, it might be beneficial to first produce columns and then load these columns when necessary.
To streamline this, you can set these dependencies as part of the Producer instance that creates the category ids (i.e. via the `category_ids` producer).

To properly set this up, add the category instances as part of the `Producer.init`. Columns from
custom requirements can be added to a Producer via the `Producer.requires` in combination
with the `Producer.setup`.
```